pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -73,6 +73,7 @@
   private final List<String> _invertedIndexCreationColumns = new ArrayList<>();
   private final List<String> _textIndexCreationColumns = new ArrayList<>();
   private final List<String> _fstIndexCreationColumns = new ArrayList<>();
+  private final List<String> _nativeFSTIndexCreationColumns = new ArrayList<>();
```

Comment:
I'd suggest adding an enum for the FST type (can be `LUCENE` or `NATIVE` instead of keeping separate lists

Reply:
Would that not limit the ability to create per column FST indices within same table?